src/main/kotlin/edu/carroll/polyominoes/service/game/GameServiceImpl.kt

```diff
@@ -0,0 +1,149 @@
+package edu.carroll.polyominoes.service.game
+
+import edu.carroll.polyominoes.jpa.model.Account
+import edu.carroll.polyominoes.jpa.model.Leaderboard
+import edu.carroll.polyominoes.jpa.repo.AccountRepository
+import edu.carroll.polyominoes.jpa.repo.LeaderboardRepo
+import edu.carroll.polyominoes.web.rest.ajax.GameStats
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.time.Clock
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+@Service
+class GameServiceImpl(private val leaderboardRepo: LeaderboardRepo, private val accountRepo: AccountRepository) :
+        GameService {
+    companion object {
+        private val log = LoggerFactory.getLogger(GameServiceImpl::class.java)
+    }
+
+    /**
+     *  When given a username and GameStats saves them to the corresponding in the database for the given user.
+     *
+     * @param user: A String representing the user to link the game statistics to.
+     * @param stats: A GameStats containing game information such as.
+     *
+     * @return Returns true if the game statistics is saved; false otherwise.
+     */
+    override fun saveGameStats(username: String, stats: GameStats): Boolean {
+
+        val user = findUser(username);
+        if (user == null) {
+            log.debug("saveGameStats: found no users for '{}'", username)
+            return false
+        }
+
+        if (stats.isEmpty()) {
+            log.debug("saveGameStats: stats is empty")
+            return false
+        }
+
+        val leaderboardRecord = leaderboardRepo.findByAccount(user);
+
+        if (leaderboardRecord.size > 1) {
+            log.warn("saveGameStats: found {} leaderboard records for '{}'", leaderboardRecord.size, username)
+            return false
+        }
+
+        val leaderboard = Leaderboard(
+                stats.score, stats.level, stats.pieces, stats.rows, stats.time, LocalDateTime.now(
+                Clock.system(ZoneId.of("America/Denver"))
```

Comment:
I would store 'raw' (UTC) times in the database, rather than trying to convert everything to a single timezone. Otherwise, you WILL have issues when you start getting data from systems in a different timezone.